src/Stratis.Bitcoin.Features.Miner.Tests/PosMintingTest.cs

```diff
@@ -511,13 +558,16 @@ private bool WasUtxoSelectedForStaking(Network network, int chainTipHeight, int
 
             var descriptions = new List<UtxoStakeDescription>();
 
-            var utxoDescription = new UtxoStakeDescription();
-            utxoDescription.TxOut = new TxOut(new Money(100), new Mock<IDestination>().Object);
-            utxoDescription.OutPoint = new OutPoint(uint256.One, 0);
-            utxoDescription.HashBlock = chainTip.Previous.HashBlock;
+            var utxoDescription = new UtxoStakeDescription
+            {
+                TxOut = new TxOut(new Money(100), new Mock<IDestination>().Object),
+                OutPoint = new OutPoint(uint256.One, 0),
+                HashBlock = chainTip.Previous.HashBlock,
+                UtxoSet = new UnspentOutputs()
+            };
 
-            utxoDescription.UtxoSet = new UnspentOutputs();
             utxoDescription.UtxoSet.SetPrivatePropertyValue("Time", chainTip.Header.Time);
+            utxoDescription.UtxoSet.SetPrivatePropertyValue("IsCoinstake", isCoinstake);
 
             // Only add the UTXO if it would have been selected by the "GetSpendable*" methods.
```

Comment:
This piece of logic (572-576 in the right side of the diff) seems to have crept back in and should be removed I think.

The `GetUtxoStakeDescriptionsSuitableForStakingAsync` should be doing this again, so there is no need for the workaround in the test.